src/avram/model.cr

```diff
@@ -184,6 +184,79 @@ class Avram::Model
     {% COLUMNS << {name: type_declaration.var, type: data_type, nilable: nilable.id, autogenerated: autogenerated} %}
   end
 
+  macro polymorphic_belongs_to(type_declaration, autogenerated = false)
+
+    {% nilable = false %}
+    {% for type in type_declaration.type.types %}
+      {% if type.class_name == "Path" %}
+        {% nilable = true if type.names[0] == "Nil" %}
+      {% elsif type.class_name == "Union" %}
```

Comment:
You can check for type like this `type.is_a?(Path)` and `type.is_a?(Union)`. This is what we do in other places so I think it'd be good to do that here as well

Reply:
what if pass a `Nil` ? like

`String | Int32 | Nil`

this way `Nil` is a Path, that's why I'm checking... or we can choose do not support `Nil`, only `?` hahaha

wdyt?